src/fhirtypes/ElementDefinition.ts

```diff
@@ -414,6 +416,101 @@ export class ElementDefinition {
       .join('.');
   }
 
+  /**
+   * Apply the AddElementRule to this new element using the appropriate methods
+   * for specific rules for the AddElementRule's implied rules (i.e., cardinality,
+   * type constraints, and flags).
+   * @param {AddElementRule} rule - specific instance of the rule
+   * @param {Fishable} fisher - A fishable implementation for finding definitions and metadata
+   */
+  applyAddElementRule(rule: AddElementRule, fisher: Fishable): void {
+    // The constrainType() method applies a type constraint to an existing
+    // ElementDefinition.type. Since this is a new ElementDefinition, it
+    // does not yet have a 'type', so we need to assign an "initial" value
+    // that the constrainType() method can process.
+    this.type = this.initializeElementType(rule, fisher);
+    const target = this.structDef.getReferenceName(rule.path, this);
+    this.constrainType(rule, fisher, target);
+
+    this.constrainCardinality(rule.min, rule.max);
+
+    this.applyFlags(
+      rule.mustSupport,
+      rule.summary,
+      rule.modifier,
+      rule.trialUse,
+      rule.normative,
+      rule.draft
+    );
+
+    // Since 'short' and 'definition' are plain string values, directly assign the values.
+    if (!isEmpty(rule.short)) {
+      this.short = rule.short;
+    }
+    if (isEmpty(rule.definition)) {
+      if (!isEmpty(rule.short)) {
+        // According to sdf-3 in http://hl7.org/fhir/structuredefinition.html#invs,
+        // definition should be provided. Default it here if possible.
+        this.definition = rule.short;
+      }
+    } else {
+      this.definition = rule.definition;
+    }
+
+    // Add the base attribute
+    this.base = {
+      path: `${this.structDef.type}.${rule.path}`,
+      min: rule.min,
+      max: rule.max
+    };
+
+    // Add the 'ele-1' constraint attribute
+    this.constraint = [
+      {
+        key: 'ele-1',
+        severity: 'error',
+        human: 'All FHIR elements must have a @value or children',
+        expression: 'hasValue() or (children().count() > id.count())',
+        xpath: '@value|f:*|h:div',
+        source: 'http://hl7.org/fhir/StructureDefinition/Element'
+      }
+    ];
```

Comment:
I think that maybe we should consider pulling this from an actual definition.  Since the source is `Element`, maybe we should fish for `Element`, grab this constraint off its first snapshot, and copy it here (adding the `source` since that is not in the original definition of the constraint).

I'm thinking we might want to do this because if a future version of FHIR tweaks the definition, we'd automatically reflect the correct constraint definition instead of inserting the old R4 variant of it.